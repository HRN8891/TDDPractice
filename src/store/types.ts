export type IUserData = {
  fullName: string;
  firstName: string;
  lastName: string;
  email: string;
};

export type IUserApiResponse = Omit<IUserData, 'fullName'>;

export interface IAuthState {
  userData: IUserData;
}
