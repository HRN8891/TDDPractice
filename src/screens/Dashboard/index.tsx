import React, {useCallback} from 'react';
import {View, TouchableOpacity} from 'react-native';

import styles from './style';
import {storageKeys, removeSecureStorageItem} from '../../utilities/storageUtils';
// import {useAppSelector} from 'store';
import CLabel from 'components/CLabel';
import {DashboardScreenProps} from 'navigation/types';

function Dashboard({navigation}: DashboardScreenProps) {
  // const userData = useAppSelector(state => state.auth.userData);

  const onCrossPress = useCallback(() => {
    removeSecureStorageItem(storageKeys.userData)
      .then(() => {
        // remove user local data
      })
      .catch(() => {});
  }, []);

  return (
    <View style={styles.constainer}>
      <TouchableOpacity onPress={onCrossPress}>{/* Add cross button icon here */}</TouchableOpacity>
      <CLabel style={styles.textStyle}>Welcome To Dashboard!!!</CLabel>
      {/* <CLabel style={styles.textStyle}>Login as: {userData.email}</CLabel> */}
    </View>
  );
}

export default Dashboard;
