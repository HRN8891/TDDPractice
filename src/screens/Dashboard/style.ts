import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  constainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textStyle: {
    fontSize: 25,
  },
});

export default styles;
