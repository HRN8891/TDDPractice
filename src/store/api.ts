import {BaseQueryApi, createApi, FetchArgs, fetchBaseQuery} from '@reduxjs/toolkit/query/react';
import config from '../constants/config';

const baseQuery = fetchBaseQuery({
  baseUrl: config.API_BASE_URL,
  prepareHeaders: (headers, {getState}) => {
    // create header with token here :
    //
    // const token: string = getState()?.auth?.accessToken;
    // if (token) {
    //   headers.set('Authorization', `Bearer ${token}`);
    // }

    return headers;
  },
});

const api = createApi({
  reducerPath: 'api',
  baseQuery: async (args: string | FetchArgs, api: BaseQueryApi, extraOptions: object) => {
    const result = await baseQuery(args, api, extraOptions);

    if (result.error?.status === 401) {
      // logout
    }
    return result;
  },
  tagTypes: [],
  endpoints: () => ({}),
});

export default api;
