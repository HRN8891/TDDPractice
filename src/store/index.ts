import {combineReducers} from 'redux';
import {configureStore} from '@reduxjs/toolkit';
import {TypedUseSelectorHook, useDispatch, useSelector} from 'react-redux';

import api from './api';
import auth from './auth/slice';

const rootReducer = combineReducers({
  auth,
  [api.reducerPath]: api.reducer,
});

export const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware => getDefaultMiddleware().concat(api.middleware),
  devTools: __DEV__,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
