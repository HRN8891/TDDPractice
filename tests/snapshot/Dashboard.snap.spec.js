import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Dashboard from '../../src/screens/Dashboard';
import {store} from '../../src/store/index';
import {Provider} from 'react-redux';

describe('Dashboard screen', () => {
  it('Dashboard screen', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Dashboard />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
