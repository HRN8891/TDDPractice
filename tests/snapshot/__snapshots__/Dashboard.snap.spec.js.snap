// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard screen Dashboard screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 25,
      }
    }
  >
    Welcome To Dashboard!!!
  </Text>
</View>
`;
